src/snowflake/snowpark/mock/functions.py

```diff
@@ -242,12 +252,20 @@ def mock_to_date(column: ColumnEmulator, fmt: Union[ColumnEmulator, str] = None)
         if data is None:
             res.append(None)
             continue
-        if auto_detect and data.isnumeric():
-            res.append(
-                datetime.datetime.utcfromtimestamp(process_numeric_time(data)).date()
-            )
-        else:
-            res.append(datetime.datetime.strptime(data, date_format).date())
+        try:
+            if auto_detect and data.isnumeric():
+                res.append(
+                    datetime.datetime.utcfromtimestamp(
+                        process_numeric_time(data)
+                    ).date()
+                )
+            else:
+                res.append(datetime.datetime.strptime(data, date_format).date())
+        except BaseException:
```

Comment:
Is there a more specific exception?

Reply:
I'm also curious what will be the error raised here